### ISSUE TITLE
Ability to select keyboard layout

### ISSUE THREAD

**Issue by shasheene** [rescuezilla/rescuezilla#91]
Mark from the UK recently contacted me with some feedback and recommendations. One of his points was:

> When I tried to input details manually I found that I was unable to type in a ‘\’ character. The right key on a UK keyboard gives a ‘<’ character instead. That was a bit frustrating too.

I'm not sure the best way to select a keymap. Likely the easiest approach is yet another submenu of the bootloader (eg, English -> US keyboard layout -> Start Rescuezilla). This would allow all users to have the correct keymap from the GRUB bootloader all the way through to the Rescuezilla desktop itself.

There may be some language/localization subtleties and GRUB limitations (including potential lack of ability to provide a compact menu).

A temporary workaround for advanced users is to edit the GRUB boot menu item before booting.

Right now localization issues such as this seems like a relatively low priority: making the Rescuezilla frontend an application worth widely using should be done first.

**Comment by pfrouleau** [comment]
Another needed keyboard layout is "French Canadian" (fr-CA). Our layout is very similar to US, but with a few accentuated characters. fr-FR layout is AZERTY instead of QUERTY.

To be honest, I usually run my apps in English, but I wanted to see how the interface looked in French. Rescuezilla is a lot simpler to use than Clonezilla.

**Comment by 45bcdf** [comment]
I would also use the fr-CA keyboard if available.

**Comment by shasheene** [comment]
Yep, the ability to select keyboard layout will be present in the next version of Rescuezilla.

I originally wanted selection of keyboard layout in the Rescuezilla boot menu, as it would mean the correct keyboard layout during very early boot, but GRUB2 doesn't [support this](https://github.com/rescuezilla/rescuezilla/issues/177)

Then I wanted to find a lightweight desktop app for keyboard selection.

But at this point, adding a more featureful settings app makes sense since some users have also wanted the ability to [switch off power saving features](https://github.com/rescuezilla/rescuezilla/issues/173).

**Comment by shasheene** [comment]
I know a keyboard layout selection app has been highly-requested, but after trying to re-evaluate approaches like *`xfce4-settings` and `gxkb`** there is definitely no app I have found that provides the user experience I'm looking for.

My plan is to adapt the Ubuntu installer python script (setting timezone and keyboard) into the Rescuezilla as a new standalone program, but this will take a few days of work, and I need to release Rescuezilla v2.2 much sooner than that.

**Comment by Bingo600** [comment]
For a Non GUI method , i used the below in a XFCE terminal window
******************************************
Change keyboard map (X) - to DK layout
https://askubuntu.com/questions/209597/how-do-i-change-keyboards-from-the-command-line

Open terminal "Accessories --> Xfce Terminal"

Enter at the # Prompt : setxkbmap dk

******************************************

**Comment by Bingo600** [comment]
Changing ISO to show english menus with Danish (dk) keybd.
1:
I tried to edit the grub.cfg , with isomaster , but the end result were ubootable.

2: 
I decided to "hack the changes directly into the ISO image" using : Bless  (linux hex editor)

Here are my work notes: 

The below is located on the ISO in file: grub.cfg
create_menu "English"                "en"      "en_US" "console-setup/layoutcode=us bootkbd=us"

I tried to change with : isomaster , but the image was unbootable.
Now we try to hack the change in , via the Bless hexeditor.

TIP:
------ 
Press the "Insert button" to get into "Replace mode", as Bless starts in "Insert mode" 
If not doing that .... Delete "backspace" the letters to be changed, before writing the new ones

We want to search in bless as text for : bootkbd=us

And ....
 
1:
Change bootkbd=us to bootkbd=dk

2: 
Change console-setup/layoutcode=u to console-setup/layoutcode=dk

3:
Change "en" to "dk" 


We want to end with this line: 
create_menu "English"                "dk"      "en_US" "console-setup/layoutcode=dk bootkbd=dk

Where the only "english" is the : "en_US" Locale
